srcpkgs/helvum/template

```diff
@@ -0,0 +1,47 @@
+# Template file for 'helvum'
+pkgname=helvum
+version=0.3.4
+revision=1
+_pipewire_rs_version=0.4.1
+build_style=cargo
+build_helper=rust
+hostmakedepends="pkg-config clang libclang"
+makedepends="gtk4-devel pipewire-devel glib-devel"
+depends="pipewire"
+short_desc="GTK patchbay for pipewire"
+maintainer="classabbyamp <dev@kb6.ee>"
+license="GPL-3.0-only"
+homepage="https://gitlab.freedesktop.org/pipewire/helvum"
+distfiles="https://gitlab.freedesktop.org/pipewire/helvum/-/archive/${version}/helvum-${version}.tar.gz
+ https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/archive/v${_pipewire_rs_version}/pipewire-rs-v${_pipewire_rs_version}.tar.gz"
+checksum="e19054f3bcd1ba234c451134fe7f4c6afee35f108c122642c1d502fc16785c02
+ 7df384c2a98154ae2cea8e14693db92e269437ea60fbcdfb6adb589226d12629"
+
+# XXX: this should not be necessary once the next version of the pipewire crate has been released
+post_extract() {
+	echo "[patch.crates-io]" >> Cargo.toml
+	for crate in pipewire pipewire-sys libspa libspa-sys; do
+		mv ../pipewire-rs-*/$crate $crate
+		echo "$crate = { path = './$crate' }" >> Cargo.toml
+	done
+}
+
+post_install() {
+	# Normally, meson would do this, but it's simpler to use the cargo build style and do this manually
+	# This also avoids the need for makedeps/build steps that are made unnecessary by xbps' hooks
+	# On update, all meson.build files should be checked for changes
+	_app_id='org.pipewire.Helvum'
+	vinstall data/icons/$_app_id.svg 644 usr/share/icons/hicolor/scalable/apps/
+	vinstall data/icons/$_app_id-symbolic.svg 644 usr/share/icons/hicolor/symbolic/apps/
+
+	vsed -i data/$_app_id.desktop.in -e "s/@icon@/$_app_id/"
+	vinstall data/$_app_id.desktop.in 644 usr/share/applications/ $_app_id.desktop
+
+	vsed -i data/$_app_id.metainfo.xml.in -e "s/@app-id@/$_app_id/"
+	vinstall data/$_app_id.metainfo.xml.in 644 usr/share/metainfo/ $_app_id.metainfo.xml
+}
+
+do_clean() {
+	# clean up pipewire-rs
+	rm -rf ${XBPS_BUILDDIR}/pipewire-rs-*
+}
```

Comment:
We can use create_wrksrc instead.

Reply:
this works but it causes the patch to need `/helvum-0.3.4/` added to the path. see the latest pushed version.